packages/amplify-category-auth/resources/lambda-function.js

```diff
@@ -6,11 +6,26 @@ exports.handler = async function (event, context) {
     const userPoolId = event.ResourceProperties.userpoolId;
     const lambdaConfig = event.ResourceProperties.lambdaConfig;
     const config = {};
+    const cognitoclient = new aws.CognitoIdentityServiceProvider();
+    const userPoolConfig = await cognitoclient.describeUserPool({ UserPoolId: userPoolId }).promise();
+    // convert describe userPool return object to updateUserpool input
+    const userPoolParams = userPoolConfig.UserPool;
+    delete userPoolParams.Id;
+    delete userPoolParams.Name;
+    delete userPoolParams.LastModifiedDate;
+    delete userPoolParams.CreationDate;
+    delete userPoolParams.SchemaAttributes;
+    delete userPoolParams.EstimatedNumberOfUsers;
+    delete userPoolParams.UsernameConfiguration;
+    delete userPoolParams.Arn;
+    delete userPoolParams.AdminCreateUserConfig.UnusedAccountValidityDays;
```

Comment:
Would it be reasonable to build the object up from nothing instead of adding everything and then removing?

Reply:
All the settings were same , I only removed the attributes which arent required in updateUserPool function.

Reply:
I agree with John, build up what we do need to pass and don't rely on what is there as if Cognito adds a new field to describe then we need to maintain this delete list.